### PR TITLE
Special chapter

### PR DESCRIPTION
**Pull request by RaphaelS1**

### CODE REVIEW

**Comment by RaphaelS1** [comment]
@damirpolat I'd appreciate your feedback/edits on the clustering section, especially there's some parts I've written badly about PCA & measures

**Comment by damirpolat** [comment]
I added a couple of things to clustering section:
- replaced some examples to use Ruspini dataset because it's much easier to visualize and perform clustering on
- added notes on measures (dbscan vs. k-means)

File: book/special.qmd [hunk 1]
```diff
@@ -674,13 +674,15 @@ In the code below we construct a cluster task using `r ref("as_task_clust")` whi
...
+library(cluster)
+task = as_task_clust(ruspini)
 task
 
 autoplot(task)
 ```
 
-Currently we only have one clustering task shipped with `r mlr3cluster`:
+Technically, we did not need to create a new task for ruspini dataset since it is already included in the package.
+Currently we only have two clustering tasks shipped with `r mlr3cluster`:
```
Comment (RaphaelS1) [inline book/special.qmd:685]:
```suggestion
Currently we have two clustering tasks shipped with `r mlr3cluster`:
```

**Review by RaphaelS1 — COMMENTED** [review]
Thanks!

File: book/special.qmd [hunk 2]
```diff
@@ -742,9 +745,10 @@ There are a few built-in evaluation metrics available to assess the quality of c
 mlr_measures$keys("clust")
 ```
 
-Two common measures are the within sum of squares (WSS) measure, `r ref("mlr_measures_clust.wss")`, and silhouette quality index (SQI), `r ref("mlr_measures_clust.silhouette")`.
+Two common measures are the within sum of squares (WSS) measure, `r ref("mlr_measures_clust.wss")`, and silhouette coefficient, `r ref("mlr_measures_clust.silhouette")`.
 WSS calculates the sum of squared differences between observations and centroids, which is a quantification of cluster cohesion (smaller values indicate clusters more compact).
-SQI quantifies how well each point belongs to its assigned cluster versus neighboring clusters, where scores closer to `1` indicate well clustered and scores closer to `-1` indicate poorly clustered.
+Silhouette coefficient quantifies how well each point belongs to its assigned cluster versus neighboring clusters, where scores closer to `1` indicate well clustered and scores closer to `-1` indicate poorly clustered.
+Note that `r mlr3cluster`'s silhouette measure outputs mean Silhouette score across all points and when there is only a single cluster, the measure simply outputs 0.
```
[the remark this answers is not in the document]
Comment (damirpolat) [reply book/special.qmd]: I meant observation. I can change it!

File: book/special.qmd [hunk 3]
```diff
@@ -754,50 +758,93 @@ meas = msrs(c("clust.wss", "clust.silhouette"))
...
 ```
 
-The very high WSS and middling SQI indicate that our clusters could do with a bit more work.
+The very high WSS and middling mean silhouette coefficient indicate that our clusters could do with a bit more work.
 Often reducing an unsupervised task to a quantitative measure may not be useful (given no ground truth) and instead visualisation may be a more effective tool for assessing the quality of the clusters.
 
+#### Measure Limitations
+It is easy to rely on clustering measures to assess the quality of clustering. 
+However this should be done with some care.
+Let's look at the following dataset.
```
[the remark this answers is not in the document]
Comment (damirpolat) [reply book/special.qmd]: Ah makes sense!

**Comment by damirpolat** [comment]
I added clarification to silhouette plot and a clustering exercise along with its solution.

File: book/special.qmd [hunk 4]
```diff
@@ -813,15 +813,17 @@ By plotting the result of PCA, we see that our model has separated observations
...
-The average silhouette index is 0.57, which means observations that are scored below this line are considered to be poor estimations whereas observations above are well clustered.
-To visualise this, silhouette plots include a dotted line at 0.57 and a bar for the score of each data point coloured by cluster.
-
-Continuing with our new example, we find that a lot of observations are actually below the average line, and therefore the quality of our cluster assignments is not very good, meaning that many observations are likely assigned to the wrong cluster.
+To visualise this, silhouette plots include a dotted line at average silhouette coefficient value and a bar for the score of each data point coloured by cluster.
+In our particular case, the average silhouette index is `r round(msr("clust.silhouette")$score(prediction, task), 2)`.
+If many or all data points in a cluster are scored below the average silhouette coefficient line or are near zero then the these observations are considered to be poorly clustered whereas observations with higher silhouette values are well clustered.
```
Comment (RaphaelS1) [inline book/special.qmd:824]: I think intuitively this is still complicated. You are saying a cluster is 'poor' based on the average which itself will depend on those 'poor' clusters! I wonder if something like below taken from [this link](https://www.scikit-yb.org/en/latest/api/cluster/silhouette.html) could be useful?

> In SilhouetteVisualizer plots, clusters with higher scores have wider silhouettes, but clusters that are less cohesive will fall short of the average score across all clusters, which is plotted as a vertical dotted red line.

File: book/special.qmd [hunk 5]
```diff
@@ -841,7 +841,7 @@ learners = list(
   lrn("clust.cmeans", centers = 3L)
 )
 measures = list(msr("clust.wss"), msr("clust.silhouette"))
-bmr = benchmark(benchmark_grid(task, learners, rsmp("cv", folds = 3)))
+bmr = benchmark(benchmark_grid(task, learners, rsmp("insample")))
```
Comment (RaphaelS1) [inline book/special.qmd:844]: why insample?

File: book/special.qmd [hunk 6]
```diff
@@ -1054,7 +1054,7 @@ We will run `set.seed(11)` before each of our solutions so you can reproduce our
...
 1. Train and predict a survival forest using `rfsrc` (from `mlr3extralearners`). Run this experiment using `task = tsk("rats"); split = partition(task)`. Evaluate your model with the RCLL measure.
 1. Estimate the density of the `tsk("precip")` data using `logspline` (from `mlr3extralearners`). Run this experiment using `task = tsk("precip"); split = partition(task)`. Evaluate your model with the logloss measure.
-1. Run a benchmark experiment on the `wine` dataset without a label column. Compare the performance of `k-means` learner with `k` equal to 2, 3, 4 using the silhouette measure. Use insample resampling technique. What value of `k` would you choose based on the silhouette scores?
+1. Run a benchmark clustering experiment on the `wine` dataset without a label column. Compare the performance of `k-means` learner with `k` equal to 2, 3 and 4 using the silhouette measure. Use insample resampling technique. What value of `k` would you choose based on the silhouette scores?
 
 <!-- TODO -->
```
Comment (RaphaelS1) [inline book/special.qmd:1059]:
```suggestion
```

File: book/special.qmd (hunk 4 above)
Comment (damirpolat) [reply book/special.qmd]: Addressed in the last commit. Thank you!

**Pull request merged**